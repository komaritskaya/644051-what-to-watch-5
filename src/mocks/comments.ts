import nanoid from 'nanoid';
import {
  getRandomNumberWithDecimals,
  getRandomDate,
  getSingleRandomItemFromArray
} from '../utils';
import {Comment} from '../types';

const USERS = [
  `Tim Macoveev`,
  `John Doe`,
  `Elon Musk`,
  `Boris Yeltsin`,
  `Johnny Knoxville`,
  `Che Guevara`,
  `Che Burashka`,
];

const PARAGRAPHS = [
  `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director’s funniest and most exquisitely designed movies in years.`,
  `Anderson’s films are too precious for some, but for those of us willing to lose ourselves in them, they’re a delight. «The Grand Budapest Hotel» is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
  `I didn’t find it amusing, and while I can appreciate the creativity, it’s an hour and 40 minutes I wish I could take back.`,
  `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
];

export const generateComment = (): Comment => ({
  id: nanoid(),
  user: getSingleRandomItemFromArray(USERS),
  date: getRandomDate(new Date(2019, 1, 1), new Date()),
  text: getSingleRandomItemFromArray(PARAGRAPHS),
  rate: getRandomNumberWithDecimals(0, 10, 1),
});

export const generateComments = (count) => {
  return new Array(count)
    .fill(``)
    .map(generateComment);
};
