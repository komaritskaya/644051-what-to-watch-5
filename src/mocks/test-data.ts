import moment from 'moment';

export const movies = [
  {
    id: `123`,
    title: `Some Film`,
    poster: `some-poster.jpg`,
    genre: `comedy`,
    year: 2000,
    director: `Some Director`,
    cast: [`Actor One`, `Actor Two`],
    cover: `some-bg-poster.jpg`,
    description: `Damn good film`,
    trailer: `some-trailer.mp4`,
    duration: moment.duration(100, `minutes`),
    comments: [
      {
        id: `111`,
        user: `User`,
        date: new Date(),
        text: `good film`,
        rate: 9.0,
      },
      {
        id: `222`,
        user: `Another User`,
        date: new Date(),
        text: `bad film`,
        rate: 1.0,
      },
    ],
  },
  {
    id: `456`,
    title: `Another Film`,
    poster: `another-poster.jpg`,
    genre: `drama`,
    year: 2000,
    director: `Another Director`,
    cast: [`Actor One`, `Actor Two`],
    cover: `another-bg-poster.jpg`,
    description: `Awful film`,
    duration: moment.duration(60, `minutes`),
    trailer: `another-trailer.mp4`,
    comments: [],
  },
];

export const tabs = [
  {
    name: `Overview`,
    link: `/`,
  },
  {
    name: `Details`,
    link: `/details`
  },
  {
    name: `Reviews`,
    link: `/reviews`,
  }
];
