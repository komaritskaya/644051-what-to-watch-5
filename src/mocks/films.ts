import nanoid from 'nanoid';
import moment from 'moment';
import {
  getRandomInt,
  getSingleRandomItemFromArray,
  getMultipleRandomItemsFromArray
} from '../utils';
import {generateComments} from './comments';
import {Movie} from '../types';

const MOVIES_COUNT = 8;

const TITLES = [
  `Avatar`,
  `Aviator`,
  `The Grand Budapest Hotel`,
  `Bohemian Rhapsody`,
  `Dardjeeling Limited`,
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Johnny English`,
  `Macbeth`,
  `Midnight Special`,
  `Mindhunter`,
  `Moonrise Kingdom`,
  `No Country for Old Men`,
  `Orlando`,
  `Player Poster`,
  `Pulp Fiction`,
  `Revenant`,
  `Seven Years in Tibet`,
  `Shutter Island`,
  `Snatch`,
  `The Grand Budapest Hotel`,
  `War of the Worlds`,
  `We Need to Talk About Kevin`,
  `What We Do in the Shadows`,
];

const POSTERS = [
  `aviator.jpg`,
  `bohemian-rhapsody.jpg`,
  `dardjeeling-limited.jpg`,
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `johnny-english.jpg`,
  `macbeth.jpg`,
  `midnight-special.jpg`,
  `mindhunter.jpg`,
  `moonrise-kingdom.jpg`,
  `no-country-for-old-men.jpg`,
  `orlando.jpg`,
  `player-poster.jpg`,
  `pulp-fiction.jpg`,
  `revenant.jpg`,
  `seven-years-in-tibet.jpg`,
  `shutter-island.jpg`,
  `snatch.jpg`,
  `the-grand-budapest-hotel-poster.jpg`,
  `war-of-the-worlds.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
];

const GENRES = [
  `Musical`,
  `Western`,
  `Drama`,
  `Comedy`,
  `Cartoon`,
  `Mystery`,
  `Horror`,
];

const DIRECTORS = [
  `Anthony Mann`,
  `Woody Allen`,
  `Pedro Almodovar`,
  `Martin Scorsese`,
  `David Lynch`,
  `Alfred Hitchcock`,
  `Takeshi Kitano`,
];

const ACTORS = [
  `Erich von Stroheim`,
  `Mary Beth Hughes`,
  `Dan Duryea`,
  `Jennifer Aniston`,
  `Courteney Cox`,
  `Lisa Kudrow`,
  `Matt LeBlanc`,
  `Matthew Perry`,
  `David Schwimmer`,
];

const COVER = `bg-the-grand-budapest-hotel.jpg`;

const DESCRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\nGustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
`;

const TRAILER = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;

const generateDurationInMinutes = (min, max) => {
  const minutes = moment.duration(getRandomInt(min, max), `minutes`);
  return minutes;
};

const generateMovie = (): Movie => ({
  id: nanoid(),
  title: getSingleRandomItemFromArray(TITLES),
  poster: getSingleRandomItemFromArray(POSTERS),
  cover: COVER,
  genre: getSingleRandomItemFromArray(GENRES),
  director: getSingleRandomItemFromArray(DIRECTORS),
  cast: getMultipleRandomItemsFromArray(ACTORS),
  year: getRandomInt(1980, 2020),
  description: DESCRIPTION,
  duration: generateDurationInMinutes(60, 180),
  trailer: TRAILER,
  comments: generateComments(getRandomInt(5, 10)),
});

const generateMovies = (count): Movie[] => {
  return new Array(count)
    .fill(``)
    .map(generateMovie);
};

export const movies = generateMovies(MOVIES_COUNT);

export const currentMovie = getSingleRandomItemFromArray(movies);
