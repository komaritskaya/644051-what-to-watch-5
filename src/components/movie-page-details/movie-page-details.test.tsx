import React from 'react';
import renderer from 'react-test-renderer';
import MoviePageDetails from './movie-page-details';
import {movies} from '../../mocks/test-data';

it(`Render Movie Page Details`, () => {
  const tree = renderer
    .create(<MoviePageDetails
      movie={movies[0]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
