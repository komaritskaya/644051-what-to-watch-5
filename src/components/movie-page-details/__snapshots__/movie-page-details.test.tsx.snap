// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movie Page Details 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Some Director
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Actor One
Actor Two
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 40m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        comedy
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2000
      </span>
    </p>
  </div>
</div>
`;
